Chap_API_Storage.tex

```diff
@@ -0,0 +1,225 @@
+Lustre storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_GPFS}
+GPFS (Spectrum Scale) storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_DATAWARP}
+Datawarp storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_NFS}
+NFS storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_CEPH}
+Ceph storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_EXT4}
+Ext4 storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_XFS}
+XFS storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_TMPFS}
+tmpfs storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_UNKNOWN_FS}
+Unknown file-based storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_DAOS}
+DAOS storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_S3}
+S3 storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_GCP}
+GCP storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_UNKNOWN_OS}
+Unknown object-based storage system.
+%
+\end{constantdesc}
+
+
+\declarestruct{pmix_storage_medium_t}
+
+The \refstruct{pmix_storage_medium_t} structure is a \code{uint16_t} type that defines a set of bit-mask flags for specifying different types of storage mediums. These can be bitwise OR'd together to accommodate storage systems that mix storage medium types.
+
+\begin{constantdesc}
+%
+\declareconstitemNEW{PMIX_STORAGE_MEDIUM_UNKNOWN}
+The storage medium type is unknown.
+%
+\declareconstitemNEW{PMIX_STORAGE_MEDIUM_TAPE}
+The storage system uses tape media.
+%
+\declareconstitemNEW{PMIX_STORAGE_MEDIUM_HDD}
+The storage system uses HDDs with traditional SAS, SATA interfaces.
+%
+\declareconstitemNEW{PMIX_STORAGE_MEDIUM_SSD}
+The storage system uses SSDs with traditional SAS, SATA interfaces.
+%
+\declareconstitemNEW{PMIX_STORAGE_MEDIUM_NVME}
+The storage system uses SSDs with NVMe interface.
+%
+\declareconstitemNEW{PMIX_STORAGE_MEDIUM_PMEM}
+The storage system uses persistent memory.
+%
+\declareconstitemNEW{PMIX_STORAGE_MEDIUM_RAM}
+The storage system is volatile (e.g., tmpfs).
+%
+\end{constantdesc}
+
+\declarestruct{pmix_storage_locality_t}
+
+The \refstruct{pmix_storage_locality_t} structure is an enumerated type for specifying the locality of storage system resources. The locality is expressed relative to the process querying the attribute.
+
+\begin{constantdesc}
+%
+\declareconstitemNEW{PMIX_STORAGE_LOCALITY_NODE}
+The storage system resources are node-local.
+%
+\declareconstitemNEW{PMIX_STORAGE_LOCALITY_NETWORK}
+The storage system resources are network-local (e.g., on the same switch).
+%
+\declareconstitemNEW{PMIX_STORAGE_LOCALITY_SITE}
+The storage system resources are site-local (i.e., in the same datacenter).
+%
+\declareconstitemNEW{PMIX_STORAGE_LOCALITY_WAN}
+The storage system resources are on the WAN (i.e., not site-local).
+%
+\end{constantdesc}
+
+
+\declarestruct{pmix_storage_access_type_t}
+
+The \refstruct{pmix_storage_access_type_t} structure is an enumerated type for specifying storage system access types.
```

Comment:
Just FYI: we avoid enumerated types in the Standard itself because it makes it impossible for local sites to extend the values. I know it isn't as nice for debuggers, but it becomes a problem when someone (e.g., a researcher) identifies some option they want to add. Implementations can implement it as an enumerated type, of course (OpenPMIx doesn't for the above reason), but we don't specify it as enumerated in the Standard.

Reply:
Sounds good. I used the same language for a few of the types I added, so I can just change all of them. The first is the `pmix_storage_type_t`, which I believe we are changing to a string from the above thread. The other 2 (the access_type_t you mention, and locality_t) both can just be changed to specify an integer type and then we just define the values as constants?